pkg/k8s/apps/crud_test.go

```diff
@@ -543,3 +544,337 @@ func TestListDevModeOn(t *testing.T) {
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name: "test",
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceMemory: mustParseK8sQuantity("128Mi"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedResources: model.ResourceRequirements{},
+			shouldInherit:     false,
+		},
+		{
+			name:        "do not inherit when dev resources are not empty",
+			envVarValue: "true",
+			dev: &model.Dev{
+				Name:      "test",
+				Container: "test",
+				Resources: model.ResourceRequirements{
+					Requests: model.ResourceList{
+						v1.ResourceMemory: mustParseModelQuantity("64Mi"),
+					},
+				},
+			},
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: v1.PodTemplateSpec{
+						Spec: v1.PodSpec{
+							Containers: []v1.Container{
+								{
+									Name: "test",
+									Resources: v1.ResourceRequirements{
+										Requests: v1.ResourceList{
+											v1.ResourceMemory: mustParseK8sQuantity("128Mi"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedResources: model.ResourceRequirements{
+				Requests: model.ResourceList{
+					v1.ResourceMemory: mustParseModelQuantity("64Mi"),
+				},
+			},
+			shouldInherit: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Save original environment variable value
+			originalValue := os.Getenv(model.OktetoInheritKubernetesResourcesEnvVar)
+			defer func() {
+				if originalValue == "" {
+					os.Unsetenv(model.OktetoInheritKubernetesResourcesEnvVar)
+				} else {
+					os.Setenv(model.OktetoInheritKubernetesResourcesEnvVar, originalValue)
+				}
+			}()
+
+			// Set test environment variable
+			if tt.envVarValue == "" {
+				os.Unsetenv(model.OktetoInheritKubernetesResourcesEnvVar)
+			} else {
+				os.Setenv(model.OktetoInheritKubernetesResourcesEnvVar, tt.envVarValue)
+			}
+
+			ctx := context.Background()
+			clientset := fake.NewSimpleClientset(tt.deployment)
+
+			app := NewDeploymentApp(tt.deployment)
+			translations, err := GetTranslations(ctx, "test", tt.dev, app, false, clientset)
+
+			assert.NoError(t, err)
+			assert.NotNil(t, translations)
+
+			// Check that the main translation exists
+			mainTranslation, exists := translations[tt.deployment.Name]
+			assert.True(t, exists)
+			assert.NotNil(t, mainTranslation)
+
+			// Check the translation rule resources
+			if len(mainTranslation.Rules) > 0 {
+				if tt.shouldInherit {
+					assert.Equal(t, tt.expectedResources, mainTranslation.Rules[0].Resources)
+				} else {
+					// When not inheriting, the rule should have the original dev resources
+					assert.Equal(t, tt.dev.Resources, mainTranslation.Rules[0].Resources)
+				}
```

Comment:
@jLopezbarb Let's split this test to prevent these conditionals. They make harder to read and maintain this test